containerd/containerd.go

```diff
@@ -141,3 +148,52 @@ func MigrateOCITarToDockerV1Tar(dir, tarName string) error {
 	return nil
 
 }
+
+// ExtractFileSystem Extract the file system from tar of an image by creating a temporary dormant container instance
+func (c Containerd) ExtractFileSystem(imageTarPath string, outputTarPath string, imageName string) error {
+	// create a new client connected to the default socket path for containerd
+	client, err := containerdApi.New("/run/containerd/containerd.sock")
```

Comment:
The socket path may not always be at this location. Vessel is agnostic of socket paths as I think there is a init time value that is passed to it. Please cross check this.

Reply:
It's defined here
https://github.com/deepfence/vessel/blob/main/constants/constants.go#L14-L15

Reply:
execution would reach this point only if the runtime is containerd instead of docker. There is only one socket address defined for containerd, which gives the error failed to dial "unix:///run/containerd/containerd.sock": context deadline exceeded. Removed "unix://" prefix and created new constant